telekom-workflow-engine/src/main/java/ee/telekom/workflow/api/WorkflowFactoryImpl.java

```diff
@@ -228,7 +234,11 @@ public WorkflowFactoryImpl validateInputVariable( int id, String attribute, Clas
     }
 
     private Tree<Row> append( Type type, Integer id, Object... arguments ){
-        Element token = new Element( type, id, arguments );
+        return append( type, id, AutoRetryOnRecovery.getDefault(), arguments );
```

Comment:
This "getDefault" seems too "magical". Maybe drop it and use AutoRetryOnRecovery.TRUE instead?

Reply:
Point here was that `getDefault` is also used in `WorkItem` - so default behaviour is defined in one place.

With current implementation this value set here doesn't get propagated to the full graph, only CallAsync nodes, other nodes have it as `null`.
When `WorkItem`s are created based on `GraphItem`s, then `WorkItem` instance is "smart" enough to return default if null.
Technically better solution would probably be to propagate this value here to all nodes so that it can never be null, but then again there is no practical need for it.